core/src/main/java/com/linecorp/armeria/common/ServerCookieEncoder.java

```diff
@@ -0,0 +1,206 @@
+     */
+    public String encode(String name, String value) {
+        return encode(Cookie.of(name, value));
+    }
+
+    /**
+     * Encodes the specified {@link Cookie} into a {@code "Set-Cookie"} header value.
+     *
+     * @param cookie the {@link Cookie} to encode.
+     * @return a single {@code "Set-Cookie"} header value.
+     */
+    public String encode(Cookie cookie) {
+        final String name = requireNonNull(cookie, "cookie").name();
+        final String value = firstNonNull(cookie.value(), "");
+
+        validateCookie(name, value);
+
+        final StringBuilder buf = stringBuilder();
+
+        if (cookie.isValueQuoted()) {
+            addQuoted(buf, name, value);
+        } else {
+            add(buf, name, value);
+        }
+
+        if (cookie.maxAge() != Long.MIN_VALUE) {
+            add(buf, CookieHeaderNames.MAX_AGE, cookie.maxAge());
+            final Date expires = new Date(cookie.maxAge() * 1000 + System.currentTimeMillis());
+            buf.append(CookieHeaderNames.EXPIRES);
+            buf.append('=');
+            DateFormatter.append(expires, buf);
+            buf.append(';');
+            buf.append(HttpConstants.SP_CHAR);
+        }
+
+        final String path = cookie.path();
+        if (path != null) {
+            add(buf, CookieHeaderNames.PATH, path);
+        }
+
+        final String domain = cookie.domain();
+        if (domain != null) {
+            add(buf, CookieHeaderNames.DOMAIN, domain);
+        }
+        if (cookie.isSecure()) {
+            add(buf, CookieHeaderNames.SECURE);
+        }
+        if (cookie.isHttpOnly()) {
+            add(buf, CookieHeaderNames.HTTPONLY);
+        }
+        final String sameSite = cookie.sameSite();
+        if (sameSite != null) {
+            add(buf, "SameSite", sameSite);
+        }
+
+        return stripTrailingSeparator(buf);
+    }
+
+    /**
+     * Encodes the specified {@link Cookie}s into the {@code "Set-Cookie"} header values.
+     *
+     * @param cookies the {@link Cookie}s to encode.
+     * @return the corresponding {@code "Set-Cookie"} header values.
+     */
+    public List<String> encode(Cookie... cookies) {
+        if (requireNonNull(cookies, "cookies").length == 0) {
+            return ImmutableList.of();
+        }
+
+        final ImmutableList.Builder<String> encoded = ImmutableList.builderWithExpectedSize(cookies.length);
+        for (final Cookie c : cookies) {
+            encoded.add(encode(c));
+        }
+        return encoded.build();
+    }
+
+    /**
+     * Encodes the specified {@link Cookie}s into the {@code "Set-Cookie"} header values.
+     *
+     * @param cookies the {@link Cookie}s to encode.
+     * @return the corresponding {@code "Set-Cookie"} header values.
+     */
+    public List<String> encode(Iterable<? extends Cookie> cookies) {
+        final Iterator<? extends Cookie> cookiesIt = requireNonNull(cookies, "cookies").iterator();
```

Comment:
It's probably worth optimizing the `Collection` case as it should be much more common than an unsized `Iterable`, allowing presizing the returned list. We could have two paths, though maybe it's overkill and better to keep the code simple and either use `ImmutableList.copyOf` or `Streams.stream(cookies).map(this::encode).collect(toImmutableList())`, both should presize the returned list.

Reply:
Done!